pkg/share/manager/nextcloud/nextcloud_test.go

```diff
@@ -726,96 +726,4 @@ var _ = Describe("Nextcloud", func() {
-			am, _ := nextcloud.NewShareManager(&nextcloud.ShareManagerConfig{
-				EndPoint: "http://mock.com/apps/sciencemesh/",
-			}, mock)
-			receivedShare, err := am.UpdateReceivedShare(ctx, &collaboration.ShareReference{
-				Spec: &collaboration.ShareReference_Id{
-					Id: &collaboration.ShareId{
-						OpaqueId: "some-share-id",
-					},
-				},
-			},
-				&collaboration.UpdateReceivedShareRequest_UpdateField{
-					Field: &collaboration.UpdateReceivedShareRequest_UpdateField_DisplayName{
-						DisplayName: "some new name for this received share",
-					},
-				})
-			Expect(err).ToNot(HaveOccurred())
-			Expect(*receivedShare).To(Equal(collaboration.ReceivedShare{
-				Share: &collaboration.Share{
-					Id:         &collaboration.ShareId{},
-					ResourceId: &provider.ResourceId{},
-					Permissions: &collaboration.SharePermissions{
-						Permissions: &provider.ResourcePermissions{
-							AddGrant:             true,
-							CreateContainer:      true,
-							Delete:               true,
-							GetPath:              true,
-							GetQuota:             true,
-							InitiateFileDownload: true,
-							InitiateFileUpload:   true,
-							ListGrants:           true,
-							ListContainer:        true,
-							ListFileVersions:     true,
-							ListRecycle:          true,
-							Move:                 true,
-							RemoveGrant:          true,
-							PurgeRecycle:         true,
-							RestoreFileVersion:   true,
-							RestoreRecycleItem:   true,
-							Stat:                 true,
-							UpdateGrant:          true,
-							DenyGrant:            true,
-						},
-					},
-					Grantee: &provider.Grantee{
-						Id: &provider.Grantee_UserId{
-							UserId: &userpb.UserId{
-								Idp:      "0.0.0.0:19000",
-								OpaqueId: "f7fbf8c8-139b-4376-b307-cf0a8c2d0d9c",
-								Type:     userpb.UserType_USER_TYPE_PRIMARY,
-							},
-						},
-					},
-					Owner: &userpb.UserId{
-						Idp:      "0.0.0.0:19000",
-						OpaqueId: "f7fbf8c8-139b-4376-b307-cf0a8c2d0d9c",
-						Type:     userpb.UserType_USER_TYPE_PRIMARY,
-					},
-					Creator: &userpb.UserId{
-						Idp:      "0.0.0.0:19000",
-						OpaqueId: "f7fbf8c8-139b-4376-b307-cf0a8c2d0d9c",
-						Type:     userpb.UserType_USER_TYPE_PRIMARY,
-					},
-					Ctime: &types.Timestamp{
-						Seconds:              1234567890,
-						Nanos:                0,
-						XXX_NoUnkeyedLiteral: struct{}{},
-						XXX_unrecognized:     nil,
-						XXX_sizecache:        0,
-					},
-					Mtime: &types.Timestamp{
-						Seconds:              1234567890,
-						Nanos:                0,
-						XXX_NoUnkeyedLiteral: struct{}{},
-						XXX_unrecognized:     nil,
-						XXX_sizecache:        0,
-					},
-				},
-				State: collaboration.ShareState_SHARE_STATE_ACCEPTED,
-			}))
-			Expect(called[0]).To(Equal(`POST /apps/sciencemesh/~tester/api/share/UpdateReceivedShare {"ref":{"Spec":{"Id":{"opaque_id":"some-share-id"}}},"f":{"Field":{"DisplayName":"some new name for this received share"}}}`))
```

Comment:
We deleted the test, because the implementing a test for this properly (using the fileld mask) requires changes that are out of our control. Please keep in mind that the CS3 API is not out of beta and still subject to changes. Modeling the sciencemesh app api after it may require you to sometimes change it as well.

Reply:
We'll make the corresponding change in the sciencemesh app api, no problem!
Please update the test instead of deleting it.
Is it failing in your branch now?

Reply:
Re-introduced and fixed the test in my PR to your PR :)